pkg/client/client.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"net/http"
+	"net/url"
+
+	"github.com/gravitational/fakeiot/pkg/metric"
+	"github.com/gravitational/roundtrip"
+	"github.com/gravitational/trace"
+)
+
+// New returns a new client
+func New(cfg Config) (*Client, error) {
+	if err := cfg.Check(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	var client *http.Client
+	if cfg.CACert != nil {
+		pool := x509.NewCertPool()
+		pool.AddCert(cfg.CACert)
+		client = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs: pool,
+				},
+			},
+		}
+	} else {
+		client = &http.Client{}
```

Comment:
Nit, this can be slightly simplified, by just creating the default client on line 37, and then if CACert is set replace it.